overlord/devicestate/devicemgr.go

```diff
@@ -2214,3 +2214,123 @@ func (m *DeviceManager) RemoveRecoveryKeys() error {
 
 	return secbootRemoveRecoveryKeys(recoveryKeyDevices)
 }
+
+// EncryptionRequirements
+// TODO: move to secboot/encryption.go (?)
+type EncryptionRequirements struct {
```

Comment:
This is a bit more than about requirements now, suggestion:

```
// EncryptionSupportInfo describes requirements and availabitly of full disk encryption support.
type EncryptionSupportInfo struct {
  // Disabled is set if to true encryption was furcefully disabled (e.g. via the seed pa partition), if set the rest of the struct content is not relevant.
  Disabled bool
  // StorageSafety ...
  StorageSafety asserts.StorageSafety
  // Available is set to true if encryption is available on this device
  // with the used gadget.
  Available bool

   // Type us set to the EncryptionType that can be used if Available is true.
   Type secboot.EncryptionType

  // UnvailableErr is set if the encryption support availability of
  // the this device and used gadget do not match the
  // storage safety requirements.
  UnvailableErr error
  // UnavailbleWarning describes why encryption support is not available in
  // case it is optional.
  UnavailbleWarning string
}
```

afaict StorageSafety captures alone enough information about to which extent encryption is desired/required?